aten/src/TH/generic/THTensorMath.cpp

```diff
@@ -545,7 +556,7 @@ void THTensor_(indexAdd)(THTensor *tensor, int dim, THLongTensor *index, THTenso
     {
       THTensor_(set1d)(tensor,
               index_data[i] - TH_INDEX_BASE,
-              THTensor_(get1d)(src,i) + THTensor_(get1d)(tensor,index_data[i] - TH_INDEX_BASE));
+              weighted?THTensor_(get1d)(weight,i):1.0 * THTensor_(get1d)(src,i) + THTensor_(get1d)(tensor,index_data[i] - TH_INDEX_BASE));
```

Comment:
I would have to look up the precedence of ?: vs. * to figure out whether this is good or not. Maybe add  `()`?

Reply:
Oh, yes `()` would make more sense.